core/Fetch.hpp

```diff
@@ -99,16 +117,45 @@ namespace olympia
         // Number of credits from decode that fetch has
         uint32_t credits_inst_queue_ = 0;
 
-        // Unit's clock
-        const sparta::Clock * my_clk_ = nullptr;
+        // Number of credits available in the ICache
+        uint32_t credits_icache_ = 0;
+
+        // Amount to left shift an Instructions PC to get the ICache block number
+        const uint32_t icache_block_shift_;
+
+        // Buffers up instructions read from the tracefile
+        std::deque<InstPtr> ibuf_;
+
+        // Size of trace buffer (must be sized >= L1ICache bandwidth / 2B)
+        const uint32_t ibuf_capacity_;
+
+        // Fetch buffer: Holds a queue of instructions that are either
+        // waiting for an ICache hit response, or they're ready to be
+        // send to decode
+        std::deque<InstPtr> fetch_buffer_;
```

Comment:
Consider using a `sparta::Queue` or `sparta::Buffer` here to get free pipeline collection, capacity checks, and utilization stats.